### ISSUE TITLE
Gradle plugins

### ISSUE THREAD

**Issue by TWiStErRob** [tomasbjerre/gradle-scripts#6]
Really interesting approach with:
```
apply from: project.buildscript.classLoader.getResource('main.gradle').toURI()
```

I wonder what was the reason behind not going with [Gradle plugins](https://docs.gradle.org/current/userguide/custom_plugins.html) and a typed DSL (Extension)?

**Comment by tomasbjerre** [comment]
The kind of code I have in main.gradle is the kind of code I find when I google things.

The kind of code I write in plugins is, in my opinion, harder. I find it harder to find documentation.

When stepping Gradle version it is also my experience that the main.gradle is easier to adjust to new Gradle versions compared to adjusting plugin code.

But I am definitely not a Gradle expert! :)

**Comment by TWiStErRob** [comment]
Fair enough. I maintain [my own plugin](https://github.com/TWiStErRob/net.twisterrob.gradle) as well, but I take the hit of "hard" to make sure that I get the deprecation warnings and compile errors as soon as possible. I've [lost faith in Groovy](https://github.com/TWiStErRob/net.twisterrob.gradle/blob/master/docs/development.md#used-languages) long time ago.

If you're interested, I would be happy to give you a tour of Gradle/Plugins through examples in your `main.gradle`. Even if you won't convert it to a plugin/Kotlin it could be an interesting pairing I think.
